android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java

```diff
@@ -516,6 +516,8 @@ public void OnWalletInitialized(int error_code) {
         }
       }
     } else if (error_code == BraveRewardsNativeWorker.WALLET_CREATED) { // Wallet created code
+      // Make sure that flag is set as panel can be closed before wallet is created
+      BraveAdsNativeHelper.nativeSetAdsEnabled(Profile.getLastUsedProfile());
```

Comment:
@samartnik is this just a front end flag or does this function goes into service and then shared lib to enable ads?

Reply:
it makes call `ads_service_->SetEnabled`

Reply:
@tmancey I think this should be avoided as UI calls should only be called based on user actions and not based on the observers